Description: Disable examples and unavailable deps
 termwiz is proper unavailable,
 font8x8 is cruft,
 document-features is useless without feature comments,
 color-eyre (many), fakeit (1) and octocrab (1) are used by examples.
 Examples are too many and drastically slow down builds and autopkgtest runs.
Forwarded: not-needed
Last-Update: 2024-08-31
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,233 +64,233 @@
 path = "src/lib.rs"
 bench = false
 
-[[example]]
-name = "async"
-path = "examples/async.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "barchart"
-path = "examples/barchart.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "barchart-grouped"
-path = "examples/barchart-grouped.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "block"
-path = "examples/block.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "calendar"
-path = "examples/calendar.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "widget-calendar",
-]
-
-[[example]]
-name = "canvas"
-path = "examples/canvas.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "chart"
-path = "examples/chart.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "colors"
-path = "examples/colors.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "colors_rgb"
-path = "examples/colors_rgb.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "palette",
-]
-
-[[example]]
-name = "constraint-explorer"
-path = "examples/constraint-explorer.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "constraints"
-path = "examples/constraints.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "custom_widget"
-path = "examples/custom_widget.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "demo"
-path = "examples/demo/main.rs"
-doc-scrape-examples = false
-
-[[example]]
-name = "demo2"
-path = "examples/demo2/main.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "palette",
-    "widget-calendar",
-]
-
-[[example]]
-name = "docsrs"
-path = "examples/docsrs.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "flex"
-path = "examples/flex.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "gauge"
-path = "examples/gauge.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "hello_world"
-path = "examples/hello_world.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "hyperlink"
-path = "examples/hyperlink.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "inline"
-path = "examples/inline.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "layout"
-path = "examples/layout.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "line_gauge"
-path = "examples/line_gauge.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "list"
-path = "examples/list.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "minimal"
-path = "examples/minimal.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "modifiers"
-path = "examples/modifiers.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "panic"
-path = "examples/panic.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "paragraph"
-path = "examples/paragraph.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "popup"
-path = "examples/popup.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "ratatui-logo"
-path = "examples/ratatui-logo.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "scrollbar"
-path = "examples/scrollbar.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "sparkline"
-path = "examples/sparkline.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "table"
-path = "examples/table.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "tabs"
-path = "examples/tabs.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "tracing"
-path = "examples/tracing.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "user_input"
-path = "examples/user_input.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "widget_impl"
-path = "examples/widget_impl.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "unstable-widget-ref",
-]
+#[[example]]
+#name = "async"
+#path = "examples/async.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "barchart"
+#path = "examples/barchart.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "barchart-grouped"
+#path = "examples/barchart-grouped.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "block"
+#path = "examples/block.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "calendar"
+#path = "examples/calendar.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "widget-calendar",
+#]
+
+#[[example]]
+#name = "canvas"
+#path = "examples/canvas.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "chart"
+#path = "examples/chart.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "colors"
+#path = "examples/colors.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "colors_rgb"
+#path = "examples/colors_rgb.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "palette",
+#]
+
+#[[example]]
+#name = "constraint-explorer"
+#path = "examples/constraint-explorer.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "constraints"
+#path = "examples/constraints.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "custom_widget"
+#path = "examples/custom_widget.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "demo"
+#path = "examples/demo/main.rs"
+#doc-scrape-examples = false
+
+#[[example]]
+#name = "demo2"
+#path = "examples/demo2/main.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "palette",
+#    "widget-calendar",
+#]
+
+#[[example]]
+#name = "docsrs"
+#path = "examples/docsrs.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "flex"
+#path = "examples/flex.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "gauge"
+#path = "examples/gauge.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "hello_world"
+#path = "examples/hello_world.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "hyperlink"
+#path = "examples/hyperlink.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "inline"
+#path = "examples/inline.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "layout"
+#path = "examples/layout.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "line_gauge"
+#path = "examples/line_gauge.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "list"
+#path = "examples/list.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "minimal"
+#path = "examples/minimal.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "modifiers"
+#path = "examples/modifiers.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "panic"
+#path = "examples/panic.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "paragraph"
+#path = "examples/paragraph.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "popup"
+#path = "examples/popup.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "ratatui-logo"
+#path = "examples/ratatui-logo.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "scrollbar"
+#path = "examples/scrollbar.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "sparkline"
+#path = "examples/sparkline.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "table"
+#path = "examples/table.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "tabs"
+#path = "examples/tabs.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "tracing"
+#path = "examples/tracing.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "user_input"
+#path = "examples/user_input.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "widget_impl"
+#path = "examples/widget_impl.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "unstable-widget-ref",
+#]
 
 [[test]]
 name = "backend_termion"
@@ -349,10 +349,10 @@
 name = "widgets_tabs"
 path = "tests/widgets_tabs.rs"
 
-[[bench]]
-name = "main"
-path = "benches/main.rs"
-harness = false
+#[[bench]]
+#name = "main"
+#path = "benches/main.rs"
+#harness = false
 
 [dependencies.bitflags]
 version = "2.3"
@@ -367,9 +367,9 @@
 version = "0.28.1"
 optional = true
 
-[dependencies.document-features]
-version = "0.2.7"
-optional = true
+#[dependencies.document-features]
+#version = "0.2.7"
+#optional = true
 
 [dependencies.indoc]
 version = "2"
@@ -399,9 +399,9 @@
 version = "0.26.3"
 features = ["derive"]
 
-[dependencies.termwiz]
-version = "0.22.0"
-optional = true
+#[dependencies.termwiz]
+#version = "0.22.0"
+#optional = true
 
 [dependencies.time]
 version = "0.3.11"
@@ -420,8 +420,8 @@
 [dev-dependencies.argh]
 version = "0.1.12"
 
-[dev-dependencies.color-eyre]
-version = "0.6.2"
+#[dev-dependencies.color-eyre]
+#version = "0.6.2"
 
 [dev-dependencies.criterion]
 version = "0.5.1"
@@ -431,11 +431,11 @@
 version = "0.28.1"
 features = ["event-stream"]
 
-[dev-dependencies.fakeit]
-version = "1.1"
+#[dev-dependencies.fakeit]
+#version = "1.1"
 
-[dev-dependencies.font8x8]
-version = "0.3.1"
+#[dev-dependencies.font8x8]
+#version = "0.3.1"
 
 [dev-dependencies.futures]
 version = "0.3.30"
@@ -443,9 +443,6 @@
 [dev-dependencies.indoc]
 version = "2"
 
-[dev-dependencies.octocrab]
-version = "0.41.0"
-
 [dev-dependencies.pretty_assertions]
 version = "1.4.0"
 
@@ -496,7 +493,7 @@
     "compact_str/serde",
 ]
 termion = ["dep:termion"]
-termwiz = ["dep:termwiz"]
+#termwiz = ["dep:termwiz"]
 underline-color = ["dep:crossterm"]
 unstable = [
     "unstable-rendered-line-info",
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -257,8 +257,8 @@
 //!     frame.render_widget(paragraph, areas[3]);
 //! }
 //! ```
-#![cfg_attr(feature = "document-features", doc = "\n## Features")]
-#![cfg_attr(feature = "document-features", doc = document_features::document_features!())]
+//#![cfg_attr(feature = "document-features", doc = "\n## Features")]
+//#![cfg_attr(feature = "document-features", doc = document_features::document_features!())]
 //!
 //! [Ratatui Website]: https://ratatui.rs/
 //! [Installation]: https://ratatui.rs/installation/
